pkg/crc/machine/certs-expiry.go

```diff
@@ -0,0 +1,66 @@
+package machine
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/code-ready/crc/pkg/crc/cluster"
+	"github.com/code-ready/crc/pkg/crc/machine/bundle"
+	crcssh "github.com/code-ready/crc/pkg/crc/ssh"
+)
+
+/* Returned when a certificate is past its expiry date */
+type certsExpiredError struct {
+	expiryTime time.Time
+}
+
+const (
+	certsExpiryUrl = "For more details, read https://code-ready.github.io/crc/#troubleshooting-expired-certificates_gsg"
+)
+
+func (err certsExpiredError) Error() string {
+	return fmt.Sprintf("Instance certificates have expired, they were valid until %s\n%s", err.expiryTime.Format(time.RFC822), certsExpiryUrl)
+}
+
+/* Returned when a certificate will expire in less than 7 days */
+type certsExpiringError struct {
+	daysLeft int
+}
+
+func (err certsExpiringError) Error() string {
+	return fmt.Sprintf("Instance certificates will expire in %d days, you should consider upgrading to a newer release\n%s", err.daysLeft, certsExpiryUrl)
+}
+
+func daysTillExpiry(expiryTime time.Time) int {
+	durationTillExpiry := time.Until(expiryTime)
+	return int(durationTillExpiry.Hours() / 24)
+}
+
+func checkCertsExpiry(expiryTime time.Time) error {
+	daysLeft := daysTillExpiry(expiryTime)
+	if daysLeft < 0 {
+		return certsExpiredError{expiryTime}
+	} else if daysLeft < 7 {
+		return certsExpiringError{daysLeft}
+	}
+
+	return nil
+}
+
+func checkVMCertsExpiry(sshRunner *crcssh.SSHRunner) error {
```

Comment:
Don't really like the use of `error` here. I believe this should be `bool, error`. In which `err` is only used when an actual error occurs.

Reply:
And I did not really like having a somehow 'magic' boolean in the previous implementation ;)
Would an enum work for you?
```
const (
    CertExpired = iota,
    CertExpiringIn7Days,
)
```

Reply:
something more explicit is always better. I think an `enum` would work better in this case... as it avoid a magic value; both processing logic and determining logic can/should use the same value in that case.

Reply:
Could not fully switch to enums if I want to hold the expiry date somewhere, so hybrid approach with error/enum.